presentation/matrix_view.py

```diff
@@ -261,22 +259,26 @@ def make_cell_hover_text(self, from_agent_id, to_agent_id: AgentID, mesh: MeshRe
         distance_unit = self._config.distance_unit
         distance = calc_distance(from_agent.coords, to_agent.coords, distance_unit)
 
-        cell_hover_text = [
-            f"{from_agent.alias} -> {to_agent.alias}",
-            f"Distance: {distance:.0f} {distance_unit.value}",
-        ]
+        cell_hover_text: List[str] = []
+        cell_hover_text.append(f"{from_agent.alias} -> {to_agent.alias}")
+        cell_hover_text.append(f"Distance: {distance:.0f} {distance_unit.value}")
 
-        if conn.has_no_data():
-            cell_hover_text.append("NO DATA")
+        health = conn.latest_measurement
+        if health:
+            cell_hover_text.append(f"Latency: {self.format_health(MetricType.LATENCY, health, True)}")
```

Comment:
For future update, consider looping over entries in the `MetricType` enum:
```
for m in MetricType:
   cell_hover_text.append(f"{m.value}: {self.format_health(m, health, True)}")
```
(this would though require to either remove units from `MetricType` values or call `format_health` with `include_unit` set to `False` to avoid having the unit in the text twice :-)).

Reply:
... and next time you can use the same approach to also fill the drop-down box for metric selection (lines 78 - 84):
```
                                dcc.Dropdown(
                                    id=self.METRIC_SELECTOR,
                                    options=[{"label": f"{m.value} [{m.unit}]", "value": m.value} for m in MetricType],
```
:-)